### PR TITLE
LDAP logins via CLI read from `PINNIPED_USERNAME` and `PINNIPED_PASSWORD` env vars

### PR DESCRIPTION
**Pull request by cfryanr**
For CLI-based auth, such as with LDAP upstream identity providers, the user may use these environment variables to avoid being interactively prompted for username and password.

This does not impact browser-based auth, such as with an OIDC upstream identity provider.

**Release note**:

```release-note
For CLI-based auth, such as with LDAP upstream identity providers, the user may use `PINNIPED_USERNAME` and `PINNIPED_PASSWORD` environment variables to avoid being interactively prompted for username and password.
```

### CODE REVIEW

**Comment by cfryanr** [comment]
I think we decided to hold off on updating the session caching logic until we have a more holistic design for supporting multiple simultaneous sessions from the same computer (logged in as different users), so I think this PR is ready for review @mattmoyer.

File: site/content/docs/howto/configure-concierge-webhook.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Configure the Pinniped Concierge to validate webhook tokens
+title: Configure the Pinniped Concierge to Validate Webhook Tokens
```
Comment (mattmoyer) [inline site/content/docs/howto/configure-concierge-webhook.md:2]: It's not hugely important to me but these were deliberately sentence case, following some common developer style guides such as [Google](https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings) and [Microsoft](https://docs.microsoft.com/en-us/style-guide/scannable-content/headings#formatting-headings)

**Review by mattmoyer — APPROVED** [review]

File: pkg/oidcclient/login.go [hunk 2]
```diff
@@ -60,6 +60,11 @@ const (
 	defaultLDAPUsernamePrompt = "Username: "
 	defaultLDAPPasswordPrompt = "Password: "
 
+	// For CLI-based auth, such as with LDAP upstream identity providers, the user may use these environment variables
+	// to avoid getting interactively prompted for username and password.
+	defaultUsernameEnvVarName = "PINNIPED_USERNAME"
+	defaultPasswordEnvVarName = "PINNIPED_PASSWORD"
```
Comment (mattmoyer) [inline pkg/oidcclient/login.go:66]: Looks like this line needs a linter ignore:

```suggestion
	defaultPasswordEnvVarName = "PINNIPED_PASSWORD" //nolint:gosec // this is not a credential
```

File: test/integration/e2e_test.go [hunk 3]
```diff
@@ -463,13 +422,82 @@ func TestE2EFullIntegration(t *testing.T) {
...
+		// Set up the username and password env vars to avoid the interactive prompts.
+		const usernameEnvVar = "PINNIPED_USERNAME"
+		originalUsername, hadOriginalUsername := os.LookupEnv(usernameEnvVar)
+		t.Cleanup(func() {
+			if hadOriginalUsername {
+				require.NoError(t, os.Setenv(usernameEnvVar, originalUsername))
+			}
+		})
+		require.NoError(t, os.Setenv(usernameEnvVar, expectedUsername))
+		const passwordEnvVar = "PINNIPED_PASSWORD"
```
Comment (mattmoyer) [inline test/integration/e2e_test.go:474]: Looks like this line also needs a linter ignore:

```suggestion
		const passwordEnvVar = "PINNIPED_PASSWORD" //nolint:gosec // this is not a credential
```

**Review by mattmoyer — COMMENTED** [review]
I saw the linter failed, but it looks like a quick fix

File: pkg/oidcclient/login.go (hunk 2 above)
Comment (cfryanr) [reply pkg/oidcclient/login.go]: Fixed in 58ab57201fd32f07ca0c29deb03a48ce312d3b58

File: test/integration/e2e_test.go (hunk 3 above)
Comment (cfryanr) [reply test/integration/e2e_test.go]: Fixed in 58ab57201fd32f07ca0c29deb03a48ce312d3b58

File: site/content/docs/howto/configure-concierge-webhook.md (hunk 1 above)
Comment (cfryanr) [reply site/content/docs/howto/configure-concierge-webhook.md]: Oops. I was just trying to make them consistent. I just made all titles and all section headers in all how-to docs use sentence case in f4829178b35eaec559b9f8c29ac92ff27444ad5b.

**Comment by cfryanr** [comment]
This passed CI. Then I force pushed to remove an empty commit which I had previously pushed to trigger CI. That caused the PR status to revert to the status of the previous commit, which had not passed CI.

Merging because it actually passed CI even though the PR doesn't show it anymore.

**Pull request merged**